server/api.go

```diff
@@ -303,6 +304,34 @@ func (a *API) connect(w http.ResponseWriter, r *http.Request) {
 	_, _ = w.Write(data)
 }
 
+func (a *API) disconnect(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodOptions {
+		return
+	}
+
+	userID := r.Header.Get("Mattermost-User-ID")
+	teamsUserID, err := a.p.store.MattermostToTeamsUserID(userID)
+	if err != nil {
+		a.p.API.LogError("The account is not connected.", "UserID", userID, "Error", err.Error())
```

Comment:
i think we should change the errors here to something like `unable to get Teams user ID from Mattermost user ID`